src/main/java/jdbc/dao/CourseDao.java

```diff
@@ -0,0 +1,139 @@
+package jdbc.dao;
+
+import jdbc.connection.ConnectionFactory;
+import jdbc.dto.CourseDto;
+import tecnodev.course.Course;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CourseDao {
+
+    private Connection connection;
+
+    public CourseDao() {
+        try{
+            this.connection = new ConnectionFactory().recuperarConexao();
+        }catch (SQLException e){
+            throw new RuntimeException("Erro ao conectar");
+        }
+    }
+
+    public void insertCourse(Course course) throws SQLException {
+        String sql = ("INSERT INTO Course " +
+                "(name, code, estimated_time_in_hours," +
+                "teacher, subcategory_id) " +
+                "VALUES (?, ?, ?, ?, ?)");
+
+        try (PreparedStatement pstm = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            connection.setAutoCommit(false);
+            pstm.setString(1, course.getName());
+            pstm.setString(2, course.getCode());
+            pstm.setInt(3, course.getEstimatedTimeInHours());
+            pstm.setString(4, course.getTeacher());
+            pstm.setLong(5, getSubCategoryId(course));
+
+            pstm.execute();
+
+            try (ResultSet rst = pstm.getGeneratedKeys()) {
+                rst.next();
+                System.out.println(rst.getLong(1));
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+            connection.commit();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            connection.rollback();
+        }
```

Comment:
Essa forma de tratar as exceções é conhecida como `try-catch cala boca`, porque mesmo que role algo, só vai ser printado e vida que segue. O ideal seria relançar a exceção, melhor ainda se for relançada como uma `RuntimeException` envelopada com um nome que faça mais sentido para o contexto. A Thais até compartilhou esse link ontem que fala um pouco sobre isso: http://tutorials.jenkov.com/java-exception-handling/exception-wrapping.html